osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -165,7 +166,11 @@ private double computeAimValue(ScoreInfo score, OsuDifficultyAttributes attribut
             aimValue *= lengthBonus;
 
             if (effectiveMissCount > 0)
-                aimValue *= calculateMissPenalty(effectiveMissCount, attributes.AimDifficultStrainCount);
+            {
+                double missedComboPercent = 1.0 - (double)score.MaxCombo / attributes.MaxCombo;
+                double estimatedSliderbreaks = Math.Min(countMeh + countOk, countMiss * attributes.AimTopWeightedSliderFactor) * DifficultyCalculationUtils.Logistic(missedComboPercent, 0.33, 15);
+                aimValue *= calculateMissPenalty(effectiveMissCount + (usingClassicSliderAccuracy ? estimatedSliderbreaks : 0), attributes.AimDifficultStrainCount);
```

Comment:
Could we instead add the `estimatedSliderbreaks` to `effectiveMissCount` where `effectiveMissCount` is calculated so everywhere else doesn't have to be aware or care about the sliderbreak calculations?

Reply:
I realise we can't because we need the aim/speed slider factors. In that case, could we move the calculation into a method that takes the required parameters (i.e the weighted slider factor)?